### PR TITLE
build: manually update PyTorch version

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
This commit sets the PyTorch and TorchVision version to nightly release 2024-12-16.

This commit adds the support for `aten.rrelu_with_noise_functional` op by decomposing it. And, also updates the existing decomposition of `aten.rrelu_with_noise` op by decomposing it to the newly added `aten.rrelu_with_noise_functional` op. It also updates the e2e tests for `aten.rrelu_with_noise` op by replacing it with its functional variant which is added here: https://github.com/pytorch/pytorch/commit/f85e23818618d43351f24e38dd7aacb40543ba0e and which captures the noise mutation which was earlier a reason for the test failures during the training mode.

This commit also removes the newly passing tests from the xfail_sets.

### CODE REVIEW

**Review by jinchen62 — APPROVED** [review]
LGTM

**Review by zjgarvey — APPROVED** [review]
Nice!
